stout/flags.cc

```diff
@@ -101,6 +188,34 @@ void Parser::Parse(int* argc, const char*** argv) {
 
     // Skip anything that doesn't look like a flag.
     if (arg.find("--") != 0) {
+      if (subcommands_.count(arg) > 0) {
+        std::cerr << "Encountered duplicate subcommand '"
+                  << arg << "'." << std::endl;
+        exit(1);
+      }
+      if (previous_subcommand_message_
+                  .value()
+                  ->GetDescriptor()
+                  ->FindFieldByName(arg)
+              != nullptr
+          && parsing_subcommand_flags_) {
+        std::cerr << "You have already set oneof 'subcommand'"
+                  << " field for the message '"
+                  << previous_subcommand_message_.value()->GetTypeName()
+                  << "'" << std::endl;
```

Comment:
I'm not following this case ... what if the subcommand has a subcommand that is the same name? For example: `./program build build` might be totally find if the first `build` has a subcommand called `build`! 

It seems like you can just ignore this and everything will still work? And then you can delete `previous_subcommand_message_` entirely? Is that correct?